crates/relayer-types/src/core/ics04_channel/packet.rs

```diff
@@ -178,6 +178,19 @@ impl Packet {
 
         height_timed_out || timestamp_timed_out
     }
+
+    /// This method will return if the memo and receiver fields are valid, e.g. not too big.
+    /// If it fails to decode the ICS20 PacketData then there is no need to validate fields
+    /// so the method will return true.
+    pub fn are_fields_valid(&self, max_memo_size: u64, max_receiver_size: u64) -> bool {
+        match serde_json::from_slice::<RawPacketData>(&self.data).map_err(Error::serde_json_error) {
+            Ok(packet) => {
+                packet.memo.len() <= max_memo_size as usize
+                    || packet.receiver.to_string().len() <= max_receiver_size as usize
```

Comment:
Shouldn't this be a conjunction?

```suggestion
                    && packet.receiver.to_string().len() <= max_receiver_size as usize
```

Reply:
Also, let's avoid an allocation here:

```suggestion
                    && packet.receiver.as_ref().len() <= max_receiver_size as usize
```

Reply:
Oh right, since I inverted the assertion it needs to be `&&`, good catch!